src/metax_api/services/file_service.py

```diff
@@ -308,6 +308,32 @@ def destroy_bulk(cls, file_identifiers):
         _logger.info('Marked %d files as deleted from project %s' % (deleted_files_count, project_identifier))
         return Response({ 'deleted_files_count': deleted_files_count }, status=status.HTTP_200_OK)
 
+    @classmethod
+    def delete_project(cls, project_id):
+        """
+        Marks files deleted, deprecates related datasets and removes all directories.
+
+        This method is called by FileRPC
+        """
+        _logger.info('Begin to delete project from database...')
+
+        file_ids = [ id for id in File.objects.filter(project_identifier=project_id).values_list('id', flat=True) ]
```

Comment:
There is a redundant loop here. `File.objects.filter(project_identifier=project_id).values_list('id', flat=True)` already gives you what you need.

I recommend trying out what following lines do:

`stuff = File.objects.filter(project_identifier=project_id)`
`stuff = File.objects.filter(project_identifier=project_id).values_list('id')`
`stuff = File.objects.filter(project_identifier=project_id).values_list('id', flat=True)`
`stuff = File.objects.filter(project_identifier=project_id).values('id')`